### ISSUE TITLE
Mobile gestureSettingsTouch

### ISSUE THREAD

**Issue by tatoosa** [openseadragon/openseadragon#1034]
hello,
i'm trying to remove all the control in smartphone devices, like touches.
I try this 

`gestureSettingsTouch: { scrollToZoom: false,
            clickToZoom:false,pinchToZoom:false,
            dblClickToZoom:false,flickEnabled:false
        }`

i check it but it doesnt work. someone can you help me
thanx for your collaboration
Simone

**Comment by avandecreme** [comment]
That should work.
Do you have any code example? What device are you trying this with?

**Comment by tatoosa** [comment]
this is the url
goo.gl/l1W1hS
i try with iphone and android smartphone...
thanx

**Comment by avandecreme** [comment]
So it seems that everything get disabled except for the pan on drag, correct?

@msalsbery Any idea?

**Comment by tatoosa** [comment]
yes all disable...

**Comment by tatoosa** [comment]
no solution ? :)

**Comment by tatoosa** [comment]
none can help me ? please

**Comment by avandecreme** [comment]
Ok, I looked at the code and it seems we are missing an option to disable panning.
Here is the relevant code: https://github.com/openseadragon/openseadragon/blob/master/src/viewer.js#L2535
I think we should add a `dragToPan` option to the gestureSettings and wrap the [those lines](https://github.com/openseadragon/openseadragon/blob/master/src/viewer.js#L2536-L2545) in a if condition.

Would you be up to make such a patch?

**Comment by tatoosa** [comment]
so adding something like this

> 
> if ( !event.preventDefaultAction && this.viewport ) {
>         gestureSettings = this.gestureSettingsByDeviceType( event.pointerType );
>         if( !this.panHorizontal ){
>             event.delta.x = 0;
>         }
>         if( !this.panVertical ){
>             event.delta.y = 0;
>         }
>         this.viewport.panBy( this.viewport.deltaPointsFromPixels( event.delta.negate() ), gestureSettings.flickEnabled );
>         if( this.constrainDuringPan ){
>             this.viewport.applyConstraints();
>         }

**Comment by avandecreme** [comment]
No, you need to add a condition inside `onCanvasDrag` like this:

```javascript
function onCanvasDrag( event ) {
    var gestureSettings;

    if ( !event.preventDefaultAction && this.viewport ) {
        gestureSettings = this.gestureSettingsByDeviceType( event.pointerType );
        if (gestureSettings.dragToPan) {
            if( !this.panHorizontal ){
                event.delta.x = 0;
            }
            if( !this.panVertical ){
                event.delta.y = 0;
            }
            this.viewport.panBy( this.viewport.deltaPointsFromPixels( event.delta.negate() ), gestureSettings.flickEnabled );
            if( this.constrainDuringPan ){
                this.viewport.applyConstraints();
            }
        }
    }

   ....
}
```

The `dragToPan` option will need to be initialized and documented. You can take example on the other gesture settings to figure out where.

**Comment by tatoosa** [comment]
something like this

```
function onCanvasDrag( event ) {
    var gestureSettings;

    if ( !event.preventDefaultAction && this.viewport ) {
        gestureSettings = this.gestureSettingsByDeviceType( event.pointerType );
        if( !this.panHorizontal ){
            event.delta.x = 0;
        }
        if( !this.panVertical ){
            event.delta.y = 0;
        }
        this.viewport.panBy( this.viewport.deltaPointsFromPixels( event.delta.negate() ), gestureSettings.flickEnabled );
        if( this.constrainDuringPan ){
            this.viewport.applyConstraints();
        }
    }


    if ( !event.preventDefaultAction && this.viewport ) {
        gestureSettings = this.gestureSettingsByDeviceType( event.pointerType );
        if (gestureSettings.dragToPan) {
            if( !this.panHorizontal ){
                event.delta.x = 0;
            }
            if( !this.panVertical ){
                event.delta.y = 0;
            }
            this.viewport.panBy( this.viewport.deltaPointsFromPixels( event.delta.negate() ), gestureSettings.flickEnabled );
            if( this.constrainDuringPan ){
                this.viewport.applyConstraints();
            }
        }
    }
```

**Comment by avandecreme** [comment]
Sorry for not being clear, you just need to add the `if (gestureSettings.dragToPan)` and the enclosing {}.
You also need to add the dragToPan option there https://github.com/openseadragon/openseadragon/blob/master/src/openseadragon.js#L1029-L1032 set to true by default. And document that option there:
https://github.com/openseadragon/openseadragon/blob/master/src/openseadragon.js#L621
https://github.com/openseadragon/openseadragon/blob/master/src/openseadragon.js#L301-L347

**Comment by tatoosa** [comment]
sorry i ask you a big favour

here's the file [https://goo.gl/jmF3DO](https://goo.gl/jmF3DO)

can you modify for me 
thanx for your collaboration

**Comment by tatoosa** [comment]
ok
i add

gestureSettingsdragToPan: { scrollToZoom: false, clickToZoom: false, dblClickToZoom: true,  pinchToZoom: true,  flickEnabled: true,  flickMinSpeed: 120, flickMomentum: 0.25, pinchRotate: false },

and this 

function onCanvasDrag( event ) {
    var gestureSettings;

    if ( !event.preventDefaultAction && this.viewport ) {
        gestureSettings = this.gestureSettingsByDeviceType( event.pointerType );
        if( !this.panHorizontal ){
            event.delta.x = 0;
        }
        if( !this.panVertical ){
            event.delta.y = 0;
        }
        this.viewport.panBy( this.viewport.deltaPointsFromPixels( event.delta.negate() ), gestureSettings.flickEnabled );
        if( this.constrainDuringPan ){
            this.viewport.applyConstraints();
        }
		
		if (gestureSettings.dragToPan) {
            if( !this.panHorizontal ){
                event.delta.x = 0;
            }
            if( !this.panVertical ){
                event.delta.y = 0;
            }
            this.viewport.panBy( this.viewport.deltaPointsFromPixels( event.delta.negate() ), gestureSettings.flickEnabled );
            if( this.constrainDuringPan ){
                this.viewport.applyConstraints();
            }
        }
		
    }

is it right?

**Comment by iangilman** [comment]
More like this: 

```
gestureSettingsMouse:   { scrollToZoom: true,  clickToZoom: true,  dblClickToZoom: false, pinchToZoom: false, flickEnabled: false, flickMinSpeed: 120, flickMomentum: 0.25, pinchRotate: false, dragToPan: true },
gestureSettingsTouch:   { scrollToZoom: false, clickToZoom: false, dblClickToZoom: true,  pinchToZoom: true,  flickEnabled: true,  flickMinSpeed: 120, flickMomentum: 0.25, pinchRotate: false, dragToPan: true },
gestureSettingsPen:     { scrollToZoom: false, clickToZoom: true,  dblClickToZoom: false, pinchToZoom: false, flickEnabled: false, flickMinSpeed: 120, flickMomentum: 0.25, pinchRotate: false, dragToPan: true },
gestureSettingsUnknown: { scrollToZoom: false, clickToZoom: false, dblClickToZoom: true, pinchToZoom: true, flickEnabled: true, flickMinSpeed: 120, flickMomentum: 0.25, pinchRotate: false, dragToPan: true },
```

...and this:

```
function onCanvasDrag( event ) {
    var gestureSettings;

    if ( !event.preventDefaultAction && this.viewport ) {
        gestureSettings = this.gestureSettingsByDeviceType( event.pointerType );
        if (gestureSettings.dragToPan) {
            if( !this.panHorizontal ){
                event.delta.x = 0;
            }
            if( !this.panVertical ){
                event.delta.y = 0;
            }
            this.viewport.panBy( this.viewport.deltaPointsFromPixels( event.delta.negate() ), gestureSettings.flickEnabled );
            if( this.constrainDuringPan ){
                this.viewport.applyConstraints();
            }
        }

    }
...
```

...and document that option there:
https://github.com/openseadragon/openseadragon/blob/master/src/openseadragon.js#L621
https://github.com/openseadragon/openseadragon/blob/master/src/openseadragon.js#L301-L347

**Comment by jonasengelmann** [comment]
I think this has already been implemented in #1956

**Comment by iangilman** [comment]
@jonasengelmann Looks like it! I shall close. @tatoosa let us know if it's still a problem.
